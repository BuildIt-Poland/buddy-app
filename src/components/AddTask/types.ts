import { RouteComponentProps } from 'react-router-dom';

export interface AddTaskProps extends RouteComponentProps {}
