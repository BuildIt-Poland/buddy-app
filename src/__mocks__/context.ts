import { AuthContextData } from 'context/AuthStore';

export const authContext = (context?: Partial<AuthContextData>) => ({
  data: { role: 'BUDDY', token: 'dummy-token', userId: '1' },
  loading: false,
  error: null,
  isAuthenticated: true,
  login: jest.fn(),
  logout: jest.fn(),
  ...context,
});
