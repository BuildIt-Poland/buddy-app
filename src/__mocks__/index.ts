export * from './general';
export * from './graphql';
export * from './schema';
