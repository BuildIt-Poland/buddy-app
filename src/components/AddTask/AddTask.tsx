import React from 'react';

const AddTask: React.FC = () => {
  return (
    <div data-testid='add-task-page'>
      <h1>Add a new task</h1>
    </div>
  );
};

export default AddTask;
