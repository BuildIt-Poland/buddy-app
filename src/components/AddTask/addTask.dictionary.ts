interface AddTask {
  ADD_TASK_TITLE: string;
  BUTTON_TEXT: string;
  ADD_TASK_PERVIEW: string;
  SUCCESS_MESSAGE: string;
  ERROR_MESSAGE: string;
  TITLE: {
    LABEL: string;
    REQUIRED: string;
  };
  DESCRIPTION: {
    LABEL: string;
    REQUIRED: string;
  };
}

const DICTIONARY: AddTask = {
  ADD_TASK_TITLE: 'New Task',
  BUTTON_TEXT: 'Add',
  ADD_TASK_PERVIEW: 'Perview',
  SUCCESS_MESSAGE: 'Task was added successfully',
  ERROR_MESSAGE: 'An error ocurred adding task',
  TITLE: {
    LABEL: 'Title',
    REQUIRED: 'Title is required',
  },
  DESCRIPTION: {
    LABEL: 'Description (HTML syntax is accepted)',
    REQUIRED: 'Description is required',
  },
};

export default DICTIONARY;
