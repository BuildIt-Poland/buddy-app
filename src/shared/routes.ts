enum ROUTES {
  BASE = '/',
  LOGIN = '/login',
  ROUTE_404 = '/404',
  BUDDY_ADD_TASK = '/buddy/newbies/:newbieId/add-task',
  BUDDY_TASK_DETAILS = '/buddy/newbies/:newbieId/tasks/:taskId',
  BUDDY_TASKS_LIST = '/buddy/newbies/:newbieId/tasks',
  BUDDY_SELECT_NEWBIE = '/buddy/newbies',
  BUDDY_ADD_NEWBIE = '/buddy/add-newbie',
  BUDDY_NEWBIE_DETAILS = '/buddy/newbies/:newbieId/details',
  NEWBIE_TASK_DETAILS = '/newbie/tasks/:taskId',
  NEWBIE_TASKS_LIST = '/newbie/tasks',
  NEWBIE_BUDDY_DETAILS = '/newbie/buddy/:buddyId/details',
}

export { ROUTES };
