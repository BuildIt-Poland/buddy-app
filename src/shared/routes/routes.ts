enum ROUTES {
  BASE = '/',
  LOGIN = '/login',
  ERROR = '/404',
  BUDDY_ADD_TASK = '/buddy/add-task',
  BUDDY_TASK_DETAILS = '/buddy/tasks/:taskId',
  BUDDY_TASKS_LIST = '/buddy/tasks',
  BUDDY_SELECT_NEWBIE = '/buddy/select-newbie',
  BUDDY_ADD_NEWBIE = '/buddy/add-newbie',
  BUDDY_NEWBIE_DETAILS = '/buddy/newbie/:id/details',
  NEWBIE_TASK_DETAILS = '/newbie/tasks/:taskId',
  NEWBIE_TASKS_LIST = '/newbie/tasks',
}

export { ROUTES };
