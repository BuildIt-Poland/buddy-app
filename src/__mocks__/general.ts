import { ROUTES } from 'shared/routes';

export const mockLocation = (path: string = ROUTES.BASE) => ({
  key: 'utwyk7',
  pathname: path,
});

export const mockSchemaTypes = (types: any = {}) => ({
  UserRole: {
    Newbie: 'NEWBIE',
    Buddy: 'BUDDY',
  },
  TaskStatus: {
    Completed: 'COMPLETED',
    Uncompleted: 'UNCOMPLETED',
  },
  ...types,
});
