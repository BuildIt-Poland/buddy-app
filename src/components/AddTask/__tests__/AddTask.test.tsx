import React from 'react';
import { create } from 'react-test-renderer';
import { MockedProvider } from '@apollo/react-testing';
import { MemoryRouter } from 'react-router';
import { ROUTES } from 'shared/routes';
import { taskDetailsMock } from '__mocks__';
import AddTask from '../AddTask';

jest.mock('@material-ui/core/CircularProgress', () => 'CircularProgress');
jest.mock('@material-ui/core/Box', () => 'Box');
jest.mock('@material-ui/core/TextField', () => 'TextField');
jest.mock('@material-ui/core/Typography', () => 'Typography');
jest.mock('components/AppWrapper', () => 'AppWrapper');
jest.mock('components/NavBar', () => 'NavBar');
jest.mock('components/RoundedButton', () => 'RoundedButton');
jest.mock('components/BackgroundShape', () => 'BackgroundShape');
jest.mock('decorators/withSnackBar', () => (component: React.FC<any>) => component);

describe('Component - AddTask', () => {
  test('renders correctly', () => {
    const component = create(
      <MockedProvider mocks={taskDetailsMock} addTypename={false}>
        <MemoryRouter initialEntries={[ROUTES.BASE]}>
          <AddTask />
        </MemoryRouter>
      </MockedProvider>
    );

    expect(component.toJSON()).toMatchSnapshot();
  });
});
