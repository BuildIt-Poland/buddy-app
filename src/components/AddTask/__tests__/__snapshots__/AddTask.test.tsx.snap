// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component - AddTask renders correctly 1`] = `
<div
  data-testid="add-task-page"
>
  <h1>
    Add a new task
  </h1>
</div>
`;
